tests/test_ccdImage.py

```diff
@@ -0,0 +1,102 @@
+# This file is part of jointcal.
+
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Test creation and use of the CcdImage class."""
+import unittest
+
+import lsst.utils.tests
+from lsst.jointcal import testUtils
+
+import lsst.afw.geom
+import lsst.jointcal
+
+
+class CcdImageTestCase(lsst.utils.tests.TestCase):
+    def setUp(self):
+        self.nStars1 = 4
+        self.nStars2 = 100
+        struct = testUtils.createTwoFakeCcdImages(num1=self.nStars1, num2=self.nStars2)
+        self.ccdImage1 = struct.ccdImageList[0]
+        self.ccdImage2 = struct.ccdImageList[1]
+        self.bbox = struct.bbox
+
+        self.associations = lsst.jointcal.Associations(struct.ccdImageList)
+
+    def checkCountStars(self, ccdImage, nStars):
+        """Check that ccdImage.countStars() is correct under various conditions
+
+        Parameters
+        ----------
+        ccdImage : `lsst.jointcal.CcdImage`
+            The ccdImage to test.
+        nStars : `int`
+            Number of stars in ccdImage's catalog.
+
+        Notes
+        -----
+        Does not test the case where some ``measuredStars`` are not ``valid``,
+        as there is no interface for doing so at the python level.
+        """
+        # By default there are no stars because catalogForFit is uninitialized.
+        measStars, refStars = ccdImage.countStars()
+        self.assertEqual(measStars, 0)
+        self.assertEqual(refStars, 0)
+
+        # With no associations, the catalog should have exactly as many valid
+        # measuredStars as were created.
+        ccdImage.resetCatalogForFit()
+        measStars, refStars = ccdImage.countStars()
+        self.assertEqual(measStars, nStars)
+        self.assertEqual(refStars, 0)
+
+        # Cross match catalogs: there will still be no refcat matches.
+        matchCut = 3.0 * lsst.afw.geom.arcseconds
+        self.associations.setCommonTangentPoint()
+        self.associations.associateCatalogs(matchCut)
+        self.assertEqual(measStars, nStars)
+        self.assertEqual(refStars, 0)
```

Comment:
How can one tell that calling setCommonTangentPoint and associateCatalogs did anything?

Reply:
If they don't raise an exception, I guess.

Reply:
An offline conversation cleared this point up: I've added a couple of sanity checks around this block to show what's happening.